django/db/models/fields/related_descriptors.py

```diff
@@ -577,15 +577,14 @@ def _apply_rel_filters(self, queryset):
             Filter the queryset for the instance this manager is bound to.
             """
             db = self._db or router.db_for_read(self.model, instance=self.instance)
-            empty_strings_as_null = connections[db].features.interprets_empty_strings_as_nulls
             queryset._add_hints(instance=self.instance)
             if self._db:
                 queryset = queryset.using(self._db)
             queryset._defer_next_filter = True
             queryset = queryset.filter(**self.core_filters)
             for field in self.field.foreign_related_fields:
                 val = getattr(self.instance, field.attname)
-                if val is None or (val == '' and empty_strings_as_null):
+                if val is None or (val == '' and connections[db].features.interprets_empty_strings_as_nulls):
```

Comment:
This one should be reverted. You'll be doing extra accesses because it is now in a loop!